network/src/metrics.rs

```diff
@@ -1,155 +1,150 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-use prometheus::{default_registry, register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+use anemo_tower::callback::MakeCallbackHandler;
+use anemo_tower::callback::ResponseHandler;
+use prometheus::{
+    register_histogram_vec_with_registry, register_int_counter_vec_with_registry,
+    register_int_gauge_vec_with_registry, IntGaugeVec, Registry,
+};
+use prometheus::{HistogramVec, IntCounterVec};
 use std::sync::Arc;
-
-pub trait NetworkMetrics {
-    fn network_available_tasks(&self) -> &IntGaugeVec;
+use std::time::Instant;
+
+#[derive(Clone)]
+pub struct NetworkMetrics1 {
+    /// Counter of requests, route is a label (ie separate timeseries per route)
+    requests: IntCounterVec,
+    /// Request latency, route is a label
+    request_latency: HistogramVec,
+    concurrent_requests: IntGaugeVec,
```

Comment:
Instead of the name `concurrent`, how about `inflight` or `pending`?

the meaning of `concurrent` is less obvious to me. Like, does that mean the maximum number that the stack will allow to be concurrent? The highest concurrent number we have ever seen? etc.

Reply:
agreed, either of those are better, more descriptive names